Grid/qcd/action/gauge/GaugeImplTypes.h

```diff
@@ -156,32 +164,44 @@ template <class S, int Nrepresentation = Nc, int Nexp = 12 > class GaugeImplType
     return Hsum.real();
   }
 
-  static inline void Project(Field &U) {
-    ProjectSUn(U);
+  static inline void Project(Field &U)
+  {
+      Group::ProjectOnSpecialGroup(U);
   }
 
-  static inline void HotConfiguration(GridParallelRNG &pRNG, Field &U) {
-    Group::HotConfiguration(pRNG, U);
+  static inline void HotConfiguration(GridParallelRNG &pRNG, Field &U)
+  {
+      Group::HotConfiguration(pRNG, U);
   }
 
-  static inline void TepidConfiguration(GridParallelRNG &pRNG, Field &U) {
+  static inline void TepidConfiguration(GridParallelRNG &pRNG, Field &U)
+  {
     Group::TepidConfiguration(pRNG, U);
   }
 
-  static inline void ColdConfiguration(GridParallelRNG &pRNG, Field &U) {
+  static inline void ColdConfiguration(GridParallelRNG &pRNG, Field &U)
+  {
     Group::ColdConfiguration(pRNG, U);
   }
+
 };
 
 
 typedef GaugeImplTypes<vComplex, Nc> GimplTypesR;
 typedef GaugeImplTypes<vComplexF, Nc> GimplTypesF;
 typedef GaugeImplTypes<vComplexD, Nc> GimplTypesD;
 
+typedef GaugeImplTypes<vComplex, Nc, 12, Sp<Nc> > SpGimplTypesR;
+typedef GaugeImplTypes<vComplexF, Nc, 12, Sp<Nc> > SpGimplTypesF;
+typedef GaugeImplTypes<vComplexD, Nc, 12, Sp<Nc> > SpGimplTypesD;
+
```

Comment:
that predates life on heart... I guess it is known work! But we can ask

Reply:
Ah, sorry! I should've investigated that in the first place. The culprit is actually line 65. I'd argue that you want the  last two template parameters in the opposite order, so that you can specify the group without specifying the (likely-to-never-change) approximation of the exponential. Actually, I saw another hardcoded 12 in an exponential approximation elsewhere. If it's a template parameter here, we should dig up where that other one was and check if we want to template that there, too.